lib/alchemy/resources_helper.rb

```diff
@@ -158,5 +158,15 @@ def render_resources
     rescue ActionView::MissingTemplate
       render :partial => 'resource', :collection => resources_instance_variable
     end
+
+    # Returns all the params necessary to get you back from where you where
+    # before: the Ransack query and the current page.
+    #
+    def current_location_params
+      {
+        q: params[:q],
+        page: params[:page]
```

Comment:
Or imagine the case, where a user changes the value she searched for. Suddenly the record is not there anymore and the page size differs from before the update. This is a common pagination problem.